File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py

N,M = map(int, input().split())
arr = list(map(int, input().split()))
s = 0
e = 0
cnt = 0
sum = arr[s]
while(e < N):
    if sum == M :
        cnt +=1
        sum -= arr[s]
        s+=1
    elif sum < M:
        e+=1
        if e >= N : break
        sum += arr[e]
    else:
        sum -= arr[s]
        s+=1

print(cnt)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    static long arr[] = new long[10_000];
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String str[] = br.readLine().split(" ");
        int N = Integer.parseInt(str[0]);
        int M = Integer.parseInt(str[1]);

        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i = 0;i  < N ; i++) arr[i] = Integer.parseInt(st.nextToken());


        int start = 0;
        int end = 0;
        long sum = arr[0];
        int cnt = 0;

        while (end < N && start < N){

            if(sum == M){
                cnt ++;
            }

            if(sum < M && end < N-1){
                end+=1;
                sum += arr[end];
            }else{
                sum -= arr[start];
                start++;
            }
        }
        System.out.println(cnt);


    }
}
